core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -347,12 +350,7 @@ object SparkSubmit {
 
       // Standalone cluster only
       // Do not set CL arguments here because there are multiple possibilities for the main class
```

Comment:
You can just remove this section altogether and move the last option to `// Other options` down there as well